test/SquareRootEngineTests.cpp

```diff
@@ -0,0 +1,382 @@
+#include <gtest/gtest.h>
+
+#include "Size.hpp"
+#include "SquareRootEngine.hpp"
+#include "TestUtility.hpp"
+#include "utility.hpp"
+
+struct SquareRootEngineTest : ::testing::Test {
+  double mPrecision;
+  int stepCount;
+  int maxSpeed;
+  float speed;
+  float zero;
+  float angle;
+  SquareRootEngine engineSUT;
+
+  SquareRootEngineTest()
+      : mPrecision(0.0001),
+        stepCount(1),
+        maxSpeed(5),
+        speed(1.f),
+        zero(0.f),
+        angle(0.f),
+        engineSUT(stepCount, maxSpeed) {}
+};
+
+auto getSpeedDelta(auto speed_before, auto speed_after) { return speed_after - speed_before; }
+
+TEST_F(SquareRootEngineTest,
+       Given1StepCountAnd1UpdateAndDriveGearThenGetCurrentSpeedShouldReturnMaxSpeed) {
+  engineSUT.set_gear(Gear::Drive);
+
+  engineSUT.update();
+
+  EXPECT_NEAR(maxSpeed, engineSUT.get_current_speed(), mPrecision);
+}
+
+TEST_F(SquareRootEngineTest,
+       Given1StepCountAndMultipleUpdatesAndDriveGearThenGetCurrentSpeedShouldReturnMaxSpeed) {
+  engineSUT.set_gear(Gear::Drive);
+
+  update_many(engineSUT, 3);
+
+  EXPECT_NEAR(maxSpeed, engineSUT.get_current_speed(), mPrecision);
+}
+
+TEST_F(SquareRootEngineTest,
+       Given1StepCountAndReverseGearThenGetCurrentSpeedShouldReturnNegativeMaxSpeed) {
+  engineSUT.set_gear(Gear::Reverse);
+
+  engineSUT.update();
+
+  EXPECT_NEAR(-maxSpeed, engineSUT.get_current_speed(), mPrecision);
+}
+
+TEST_F(
+    SquareRootEngineTest,
+    Given1StepCountAndMultipleUpdatesAndReverseGearThenGetCurrentSpeedShouldReturnNegativeMaxSpeed) {
+  engineSUT.set_gear(Gear::Reverse);
+
+  update_many(engineSUT, 3);
+
+  EXPECT_NEAR(-maxSpeed, engineSUT.get_current_speed(), mPrecision);
+}
+
+TEST_F(SquareRootEngineTest, GivenNeutralGearThenGetCurrentSpeedShouldReturn0) {
+  engineSUT.set_gear(Gear::Neutral);
+
+  engineSUT.update();
+
+  EXPECT_NEAR(zero, engineSUT.get_current_speed(), mPrecision);
+}
+
+TEST_F(SquareRootEngineTest,
+       Given2StepCountAndDriveGearAndMaxSpeed1ThenGetCurrentSpeedShouldReturn07071) {
+  stepCount = 2;
+  maxSpeed = 1;
+  float expectedSpeed = 0.7071f;
+  SquareRootEngine engineSUT(stepCount, maxSpeed);
+  engineSUT.set_gear(Gear::Drive);
+
+  engineSUT.update();
+
+  EXPECT_NEAR(expectedSpeed, engineSUT.get_current_speed(), mPrecision);
+}
+
+TEST_F(SquareRootEngineTest,
+       Given3StepCountAnd3UpdatesAndDriveGearThenGetCurrentSpeedShouldReturnMaxSpeed) {
+  stepCount = 3;
+  SquareRootEngine engineSUT(stepCount, maxSpeed);
```

Comment:
Wait, you are shadowing the variable from the fixture. Something is broken here! Either remove TEST_F or don't redeclare variable! The same suggestion for other test cases.